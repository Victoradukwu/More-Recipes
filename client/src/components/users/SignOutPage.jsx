import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { logout } from '../../actions/userAction';

const SignoutPage = (props) => {
  localStorage.removeItem('token');
  props.logout();
  props.history.push('/signin');
  return <div />;
};

SignoutPage.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired,

};

const mapDispatchToProps = dispatch =>
  bindActionCreators({ logout }, dispatch);


export default connect(null, mapDispatchToProps)(SignoutPage);
