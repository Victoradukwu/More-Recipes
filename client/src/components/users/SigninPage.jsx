import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { siginUser } from '../../actions/userAction';
import validateFields from '../../helpers/validateFields';


class SigninPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      identifier: '',
      password: '',
      errors: {},
      isLoading: false
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.error.status) {
      this.setState({ errors: nextProps.error });
    } else {
      this.props.history.push('/dashboard');
    }
  }
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  onSubmit(event) {
    event.preventDefault();
    this.setState({ errors: {} });
    const userObject = {
      username: this.state.username,
      password: this.state.password,
    };
    if (validateFields(userObject)) {
      this.props.signinUser(userObject);
    } else {
      this.setState({ errors: { status: true, error: { message: 'Please fill in all required fields and submit again' } } });
    }
  }
  render() {
    return (
      <div>
        <br /><br />
        <div className="container main">
          <div className="main-login main-center" >
            <h3>Sign in to continue</h3>
            <form className="form-horizontal" onSubmit={this.onSubmit}>

              <div className="form-group">
                <div className="input-group">
                  <span className="input-group-addon"><i className="fa fa-user-circle fa-lg" aria-hidden="true" /></span>
                  <input onChange={this.onChange} type="text" className="form-control" name="username" id="username" placeholder="Enter your Username" />
                </div>
              </div>
              <div className="form-group">
                <div className="input-group">
                  <span className="input-group-addon"><i className="fa fa-lock fa-lg" aria-hidden="true" /></span>
                  <input onChange={this.onChange} type="password" className="form-control" name="password" id="password" placeholder="Enter your Password" />
                </div>
              </div>

              <div className="form-group ">
                <button type="submit" className="btn btn-lg btn-block search">Sign in</button>
              </div>
              {this.state.errors.status && <p style={{ color: 'red' }}>{this.state.errors.error.message}</p>}
              <hr />
              <p className="text-center">If you are new,
                <Link to="/signup" >Sign up</Link>, or
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}
SigninPage.prototypes = {
  signinUser: PropTypes.func.isRequired,
  history: PropTypes.any.isRequired,
  error: PropTypes.any.isRequired

};

const mapStateToProps = state => ({
  loading: state.authenticatingUser,
  error: state.authenticationFailed,
  success: state.authenticationSuccess
});

const mapDispatchToProps = dispatch => ({
  signinUser: credentials => dispatch(siginUser(credentials))
});

export default connect(mapStateToProps, mapDispatchToProps)(SigninPage);
