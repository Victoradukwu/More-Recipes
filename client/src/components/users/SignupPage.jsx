import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Spinner from 'react-md-spinner';
import { authenticateUser } from '../../actions/userAction';
import { signUpValidation } from '../../helpers/validateFields';

class SignupPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.isValid = this.isValid.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleOnFocus = this.handleOnFocus.bind(this);
  }


  componentWillReceiveProps(nextProps) {
    if (nextProps.error) {
      this.setState({ errors: nextProps.error });
    }
    if (nextProps.userId !== 0) {
      this.props.history.push('/');
    }
  }

  onChange(event) {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }
  onSubmit(event) {
    event.preventDefault();
    const {
      name, username, email, password, confirmPassword
    } = this.state;
    if (this.isValid()) {
      this.setState({ errors: {} });
      this.props.authenticateUser({
        name, username, email, password, confirmPassword
      }, 'signup');
    }
  }

  /**
   * @description handles client validation checks
   * @method isValid
   *
   * @returns { bool } true/false when form is submitted
   */
  isValid() {
    const { errors, isValid } = signUpValidation(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  /**
   * @description handles on focus event
   * @method handleOnFocus
   *
   * @param { object } event - event object containing sign up details
   *
   * @returns { object } new sign up details state
   */
  handleOnFocus(event) {
    this.setState({
      errors: Object.assign({}, this.state.errors, { [event.target.name]: '' })
    });
  }


  render() {
    const {
      name, email, username, password, confirmPassword
    } = this.state.errors;
    return (
      <div className="container main login-screen">
        <br /><br />

        <div className="main-login main-center" >
          <h3>Register to continue</h3>
          <form className="form-horizontal" onSubmit={this.onSubmit}>


            <div className="form-group">
              <div className="cols-sm-10">
                <div className="input-group">
                  <span className="input-group-addon">
                    <i className="authIcon fa fa-user fa" aria-hidden="true" />
                  </span>
                  <input
                    onChange={this.onChange}
                    onFocus={this.handleOnFocus}
                    type="text"
                    className="form-control"
                    value={this.state.name}
                    name="name"
                    id="name"
                    placeholder="Enter your Name"
                  />
                </div>
              </div>
              {name && <small style={{ color: 'red' }} >{name}</small>}
            </div>

            <div className="form-group">
              <div className="cols-sm-10">
                <div className="input-group">
                  <span className="input-group-addon">
                    <i className="authIcon fa fa-envelope fa" />
                  </span>
                  <input
                    onChange={this.onChange}
                    onFocus={this.handleOnFocus}
                    type="text"
                    className="form-control"
                    value={this.state.email}
                    name="email"
                    id="email"
                    placeholder="Enter your Email"
                  />
                </div>
              </div>
              {email && <small style={{ color: 'red' }} >{email}</small>}
            </div>

            <div className="form-group">
              <div className="cols-sm-10">
                <div className="input-group">
                  <span className="input-group-addon">
                    <i className="authIcon fa fa-user-circle fa-lg" />
                  </span>
                  <input
                    onChange={this.onChange}
                    onFocus={this.handleOnFocus}
                    type="text"
                    className="form-control"
                    value={this.state.username}
                    name="username"
                    id="username"
                    placeholder="Enter your Username"
                  />
                </div>
              </div>
              {username && <small style={{ color: 'red' }} >{username}</small>}
            </div>

            <div className="form-group">
              <div className="cols-sm-10">
                <div className="input-group">
                  <span className="input-group-addon">
                    <i className="authIcon fa fa-lock fa-lg" />
                  </span>
                  <input
                    onChange={this.onChange}
                    onFocus={this.handleOnFocus}
                    type="password"
                    className="form-control"
                    value={this.state.password}
                    name="password"
                    id="password"
                    placeholder="Enter your Password"
                  />
                </div>
              </div>
              {password && <small style={{ color: 'red' }} >{password}</small>}
            </div>

            <div className="form-group">
              <div className="cols-sm-10">
                <div className="input-group">
                  <span className="input-group-addon">
                    <i className="authIcon fa fa-lock fa-lg" />
                  </span>
                  <input
                    onChange={this.onChange}
                    onFocus={this.handleOnFocus}
                    type="password"
                    className="form-control"
                    value={this.state.confirmPassword}
                    name="confirmPassword"
                    id="confirm"
                    placeholder="Confirm your Password"
                  />
                </div>
              </div>
              {confirmPassword &&
                <small style={{ color: 'red' }} >
                  {confirmPassword}
                </small>
              }
            </div>

            <div className="form-group ">
              <button
                type="submit"
                className="btn btn-lg btn-block search"
                disabled={this.props.isCreating}
              >
                {!this.props.isCreating
                  ?
                  'Register'
                  :
                  <span>Registering <Spinner size={20} /></span>
                }
              </button>
            </div>
            {this.state.errors.status &&
            <p style={{ color: 'red' }}>
              {this.state.errors.error.message}
            </p>}
            <p>
              Already registered, please <Link to="/signin">Login.</Link>
            </p>
          </form>
        </div>
      </div>
    );
  }
}

SignupPage.propTypes = {
  error: PropTypes.object,
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired,
  authenticateUser: PropTypes.func.isRequired,
  userId: PropTypes.number.isRequired,
  isCreating: PropTypes.bool.isRequired
};
SignupPage.defaultProps = {
  error: {}
};

const mapStateToProps = state => ({
  userId: state.userAuthentication.authId,
  error: state.userAuthentication.signupError,
  isCreating: state.userAuthentication.isAuthenticating
});

const mapDispatchToProps = dispatch => ({
  authenticateUser: (userDetails, path) =>
    dispatch(authenticateUser(userDetails, path))
});

export default connect(mapStateToProps, mapDispatchToProps)(SignupPage);
