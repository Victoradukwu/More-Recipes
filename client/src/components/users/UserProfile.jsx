import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export class UserProfile extends React.Component {
  render() {
    const { user } = this.props;
    const profilePix = user.profilePicture ? user.profilePicture :
      'http://res.cloudinary.com/victoradukwu/image/upload/v1521336087' +
      '/avatar7_nt7awy.png';
    return (
      <div className="container">
        <div className="row profilebox">
          <div
            className="
            offset-lg-4 col-lg-4 col-sm-6 col-12 main-section text-center"
          >
            <div className="row">
              <div className="col-lg-12 col-sm-12 col-12 profile-header" />
            </div>
            <div className="row user-detail">
              <div className="col-lg-12 col-sm-12 col-12">
                <img
                  src={profilePix}
                  className="rounded-circle img-thumbnail"
                  alt="userImage"
                />
                <h5>{user.name}</h5>
                <p>
                  <i
                    className="fa fa-map-marker"
                    aria-hidden="true"
                  >{user.location}
                  </i>
                </p>

                <hr />
                Date joined: {(new Date(user.createdAt)).toLocaleDateString()}
                <br />
                Number of recipes contributed:
                {this.props.contribution}
                <br />
                <hr />
              </div>
            </div>
            <div className="row user-social-detail">
              <div className="col-lg-12 col-sm-12 col-12">
                <span className="fa fa-phone"> &nbsp;{user.phoneNumber} </span>
                <span className="fa fa-envelope">&nbsp;{user.email}</span>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

UserProfile.propTypes = {
  user: PropTypes.object.isRequired,
  contribution: PropTypes.number
};
UserProfile.defaultProps = {
  contribution: 0
};

const mapStateToProps = state => ({
  user: state.userAuthentication.user,
  contribution: state.userAuthentication.contribution
});

export default connect(mapStateToProps, null)(UserProfile);
