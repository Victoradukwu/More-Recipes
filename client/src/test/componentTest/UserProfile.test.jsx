import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { UserProfile } from '../../components/users/UserProfile';

const props = {
  user: {
    recipes: []
  }
};

describe('UserProfile', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});


describe('UserProfile', () => {
  it('Should render user profile Page Correctly', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    expect(wrapper.find('div').exists()).toBeTruthy();
    expect(wrapper.find('img').exists()).toBeTruthy();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
