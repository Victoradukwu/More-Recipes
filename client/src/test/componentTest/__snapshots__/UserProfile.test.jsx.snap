// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Should render user profile Page Correctly 1`] = `
<div
  className="container"
>
  <div
    className="row profilebox"
  >
    <div
      className=" offset-lg-4 col-lg-4 col-sm-6 col-12 main-section text-center"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12 col-sm-12 col-12 profile-header"
        />
      </div>
      <div
        className="row user-detail"
      >
        <div
          className="col-lg-12 col-sm-12 col-12"
        >
          <img
            alt="userImage"
            className="rounded-circle img-thumbnail"
            src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
          />
          <h5 />
          <p>
            <i
              aria-hidden="true"
              className="fa fa-map-marker"
            />
          </p>
          <hr />
          Date joined: 
          Invalid Date
          <br />
          Number of recipes contributed: 
          0
              
          <br />
          <hr />
        </div>
      </div>
      <div
        className="row user-social-detail"
      >
        <div
          className="col-lg-12 col-sm-12 col-12"
        >
          <span
            className="fa fa-phone"
          >
              
             
          </span>
          <span
            className="fa fa-envelope"
          >
             
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserProfile matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfile
    user={
      Object {
        "recipes": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row profilebox"
      >
        <div
          className=" offset-lg-4 col-lg-4 col-sm-6 col-12 main-section text-center"
        >
          <div
            className="row"
          >
            <div
              className="col-lg-12 col-sm-12 col-12 profile-header"
            />
          </div>
          <div
            className="row user-detail"
          >
            <div
              className="col-lg-12 col-sm-12 col-12"
            >
              <img
                alt="userImage"
                className="rounded-circle img-thumbnail"
                src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
              />
              <h5 />
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-map-marker"
                />
              </p>
              <hr />
              Date joined: 
              Invalid Date
              <br />
              Number of recipes contributed: 
              0
                  
              <br />
              <hr />
            </div>
          </div>
          <div
            className="row user-social-detail"
          >
            <div
              className="col-lg-12 col-sm-12 col-12"
            >
              <span
                className="fa fa-phone"
              >
                  
                 
              </span>
              <span
                className="fa fa-envelope"
              >
                 
              </span>
            </div>
          </div>
        </div>
      </div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className=" offset-lg-4 col-lg-4 col-sm-6 col-12 main-section text-center"
        >
          <div
            className="row"
          >
            <div
              className="col-lg-12 col-sm-12 col-12 profile-header"
            />
          </div>
          <div
            className="row user-detail"
          >
            <div
              className="col-lg-12 col-sm-12 col-12"
            >
              <img
                alt="userImage"
                className="rounded-circle img-thumbnail"
                src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
              />
              <h5 />
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-map-marker"
                />
              </p>
              <hr />
              Date joined: 
              Invalid Date
              <br />
              Number of recipes contributed: 
              0
                  
              <br />
              <hr />
            </div>
          </div>
          <div
            className="row user-social-detail"
          >
            <div
              className="col-lg-12 col-sm-12 col-12"
            >
              <span
                className="fa fa-phone"
              >
                  
                 
              </span>
              <span
                className="fa fa-envelope"
              >
                 
              </span>
            </div>
          </div>
        </div>,
        "className": "row profilebox",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="row"
            >
              <div
                className="col-lg-12 col-sm-12 col-12 profile-header"
              />
            </div>,
            <div
              className="row user-detail"
            >
              <div
                className="col-lg-12 col-sm-12 col-12"
              >
                <img
                  alt="userImage"
                  className="rounded-circle img-thumbnail"
                  src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
                />
                <h5 />
                <p>
                  <i
                    aria-hidden="true"
                    className="fa fa-map-marker"
                  />
                </p>
                <hr />
                Date joined: 
                Invalid Date
                <br />
                Number of recipes contributed: 
                0
                    
                <br />
                <hr />
              </div>
            </div>,
            <div
              className="row user-social-detail"
            >
              <div
                className="col-lg-12 col-sm-12 col-12"
              >
                <span
                  className="fa fa-phone"
                >
                    
                   
                </span>
                <span
                  className="fa fa-envelope"
                >
                   
                </span>
              </div>
            </div>,
          ],
          "className": " offset-lg-4 col-lg-4 col-sm-6 col-12 main-section text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-lg-12 col-sm-12 col-12 profile-header"
              />,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "col-lg-12 col-sm-12 col-12 profile-header",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-lg-12 col-sm-12 col-12"
              >
                <img
                  alt="userImage"
                  className="rounded-circle img-thumbnail"
                  src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
                />
                <h5 />
                <p>
                  <i
                    aria-hidden="true"
                    className="fa fa-map-marker"
                  />
                </p>
                <hr />
                Date joined: 
                Invalid Date
                <br />
                Number of recipes contributed: 
                0
                    
                <br />
                <hr />
              </div>,
              "className": "row user-detail",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="userImage"
                    className="rounded-circle img-thumbnail"
                    src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
                  />,
                  <h5 />,
                  <p>
                    <i
                      aria-hidden="true"
                      className="fa fa-map-marker"
                    />
                  </p>,
                  <hr />,
                  "Date joined: ",
                  "Invalid Date",
                  <br />,
                  "Number of recipes contributed: ",
                  0,
                  "    ",
                  <br />,
                  <hr />,
                ],
                "className": "col-lg-12 col-sm-12 col-12",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "userImage",
                    "className": "rounded-circle img-thumbnail",
                    "src": "http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      aria-hidden="true"
                      className="fa fa-map-marker"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "children": undefined,
                      "className": "fa fa-map-marker",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                "Date joined: ",
                "Invalid Date",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "Number of recipes contributed: ",
                0,
                "    ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-lg-12 col-sm-12 col-12"
              >
                <span
                  className="fa fa-phone"
                >
                    
                   
                </span>
                <span
                  className="fa fa-envelope"
                >
                   
                </span>
              </div>,
              "className": "row user-social-detail",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="fa fa-phone"
                  >
                      
                     
                  </span>,
                  <span
                    className="fa fa-envelope"
                  >
                     
                  </span>,
                ],
                "className": "col-lg-12 col-sm-12 col-12",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "  ",
                      undefined,
                      " ",
                    ],
                    "className": "fa fa-phone",
                  },
                  "ref": null,
                  "rendered": Array [
                    "  ",
                    undefined,
                    " ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      undefined,
                    ],
                    "className": "fa fa-envelope",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    undefined,
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row profilebox"
        >
          <div
            className=" offset-lg-4 col-lg-4 col-sm-6 col-12 main-section text-center"
          >
            <div
              className="row"
            >
              <div
                className="col-lg-12 col-sm-12 col-12 profile-header"
              />
            </div>
            <div
              className="row user-detail"
            >
              <div
                className="col-lg-12 col-sm-12 col-12"
              >
                <img
                  alt="userImage"
                  className="rounded-circle img-thumbnail"
                  src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
                />
                <h5 />
                <p>
                  <i
                    aria-hidden="true"
                    className="fa fa-map-marker"
                  />
                </p>
                <hr />
                Date joined: 
                Invalid Date
                <br />
                Number of recipes contributed: 
                0
                    
                <br />
                <hr />
              </div>
            </div>
            <div
              className="row user-social-detail"
            >
              <div
                className="col-lg-12 col-sm-12 col-12"
              >
                <span
                  className="fa fa-phone"
                >
                    
                   
                </span>
                <span
                  className="fa fa-envelope"
                >
                   
                </span>
              </div>
            </div>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className=" offset-lg-4 col-lg-4 col-sm-6 col-12 main-section text-center"
          >
            <div
              className="row"
            >
              <div
                className="col-lg-12 col-sm-12 col-12 profile-header"
              />
            </div>
            <div
              className="row user-detail"
            >
              <div
                className="col-lg-12 col-sm-12 col-12"
              >
                <img
                  alt="userImage"
                  className="rounded-circle img-thumbnail"
                  src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
                />
                <h5 />
                <p>
                  <i
                    aria-hidden="true"
                    className="fa fa-map-marker"
                  />
                </p>
                <hr />
                Date joined: 
                Invalid Date
                <br />
                Number of recipes contributed: 
                0
                    
                <br />
                <hr />
              </div>
            </div>
            <div
              className="row user-social-detail"
            >
              <div
                className="col-lg-12 col-sm-12 col-12"
              >
                <span
                  className="fa fa-phone"
                >
                    
                   
                </span>
                <span
                  className="fa fa-envelope"
                >
                   
                </span>
              </div>
            </div>
          </div>,
          "className": "row profilebox",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="row"
              >
                <div
                  className="col-lg-12 col-sm-12 col-12 profile-header"
                />
              </div>,
              <div
                className="row user-detail"
              >
                <div
                  className="col-lg-12 col-sm-12 col-12"
                >
                  <img
                    alt="userImage"
                    className="rounded-circle img-thumbnail"
                    src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
                  />
                  <h5 />
                  <p>
                    <i
                      aria-hidden="true"
                      className="fa fa-map-marker"
                    />
                  </p>
                  <hr />
                  Date joined: 
                  Invalid Date
                  <br />
                  Number of recipes contributed: 
                  0
                      
                  <br />
                  <hr />
                </div>
              </div>,
              <div
                className="row user-social-detail"
              >
                <div
                  className="col-lg-12 col-sm-12 col-12"
                >
                  <span
                    className="fa fa-phone"
                  >
                      
                     
                  </span>
                  <span
                    className="fa fa-envelope"
                  >
                     
                  </span>
                </div>
              </div>,
            ],
            "className": " offset-lg-4 col-lg-4 col-sm-6 col-12 main-section text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col-lg-12 col-sm-12 col-12 profile-header"
                />,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "col-lg-12 col-sm-12 col-12 profile-header",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col-lg-12 col-sm-12 col-12"
                >
                  <img
                    alt="userImage"
                    className="rounded-circle img-thumbnail"
                    src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
                  />
                  <h5 />
                  <p>
                    <i
                      aria-hidden="true"
                      className="fa fa-map-marker"
                    />
                  </p>
                  <hr />
                  Date joined: 
                  Invalid Date
                  <br />
                  Number of recipes contributed: 
                  0
                      
                  <br />
                  <hr />
                </div>,
                "className": "row user-detail",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="userImage"
                      className="rounded-circle img-thumbnail"
                      src="http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png"
                    />,
                    <h5 />,
                    <p>
                      <i
                        aria-hidden="true"
                        className="fa fa-map-marker"
                      />
                    </p>,
                    <hr />,
                    "Date joined: ",
                    "Invalid Date",
                    <br />,
                    "Number of recipes contributed: ",
                    0,
                    "    ",
                    <br />,
                    <hr />,
                  ],
                  "className": "col-lg-12 col-sm-12 col-12",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "userImage",
                      "className": "rounded-circle img-thumbnail",
                      "src": "http://res.cloudinary.com/victoradukwu/image/upload/v1521336087/avatar7_nt7awy.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h5",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <i
                        aria-hidden="true"
                        className="fa fa-map-marker"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-hidden": "true",
                        "children": undefined,
                        "className": "fa fa-map-marker",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "hr",
                  },
                  "Date joined: ",
                  "Invalid Date",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "Number of recipes contributed: ",
                  0,
                  "    ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "hr",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col-lg-12 col-sm-12 col-12"
                >
                  <span
                    className="fa fa-phone"
                  >
                      
                     
                  </span>
                  <span
                    className="fa fa-envelope"
                  >
                     
                  </span>
                </div>,
                "className": "row user-social-detail",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="fa fa-phone"
                    >
                        
                       
                    </span>,
                    <span
                      className="fa fa-envelope"
                    >
                       
                    </span>,
                  ],
                  "className": "col-lg-12 col-sm-12 col-12",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "  ",
                        undefined,
                        " ",
                      ],
                      "className": "fa fa-phone",
                    },
                    "ref": null,
                    "rendered": Array [
                      "  ",
                      undefined,
                      " ",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        undefined,
                      ],
                      "className": "fa fa-envelope",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      undefined,
                    ],
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
